src/orchestrator/unary.rs

```diff
@@ -845,7 +856,7 @@ pub async fn detect_for_context(
     let client = ctx
         .clients
         .get_as::<TextContextDocDetectorClient>(&detector_id)
-        .unwrap();
+        .expect("error getting text context doc detector client");
```

Comment:
I didn't really comment on the messages, but I'm thinking we should probably standardize around "x not found" instead of "error getting x", etc. throughout the code.

In this case, we do validate that a requested `detector_id` exists prior to getting to this point, so I think that is why we currently `unwrap()` here and in some other places.